bluerobotics/companion
# Create DVL service

**This work has been pushed to [it's own branch](https://github.com/bluerobotics/companion/tree/dvl)**.
This Draft PR will stay here for now as there are important instructions to follow.

![image](https://user-images.githubusercontent.com/4013804/87191746-6a8d0d80-c2cb-11ea-9499-b6443c805188.png)


TO-DO:
 - [ ] thread locks?
 - [x] save settings

to test, set the DVL ip address to 192.168.2.95, open a terminal in companion and do:

 - Start from companion 0.0.22
 - Make sure the ROV is connected to the internet
 - Navigate to http://192.168.2.2:2770/git
 - Click "origin" -> "branches" -> "dvl"
 - Click "update"
 - Wait until the update is over and the companion reboots
 - Refresh the page and "Water Linked" should show up in the navbar

Ardusub code is currently in [this branch](https://github.com/Williangalvani/ardupilot/tree/dvl_cleaned) (which needs some cleaning too) and a binary is available [~here~](https://github.com/bluerobotics/companion/files/4893717/ardusub.zip) [~here~](https://github.com/bluerobotics/companion/files/4949954/ardusub.zip) [merged to master](https://firmware.ardupilot.org/Sub/latest/Pixhawk1/ardusub.apj). 

suggested Parameters:

Use mavlink2!
```
SERIAL0_PROTOCOL 2
```
Enable odom
Use ekf3 for the odometry support
```
AHRS_EKF_TYPE 3
EK2_ENABLE 0
```
Enable odometry support
```
EK3_ENABLE 1
VISO_TYPE 1
```
Disable external gps in ekf3
```
EK3_GPS_TYPE 3
```
Suggested gains
```
PSC_POSXY_P 2.5
PSC_POSZ_P  1.0
PSC_VELXY_D 0.8
PSC_VELXY_I 0.5
PSC_VELXY_P 5.0
PSC_VELZ_P  5.0
```
